include/spdlog/details/logger_impl.h

```diff
@@ -87,6 +88,9 @@ inline void spdlog::logger::log(level::level_enum lvl, const char* msg)
     {
         details::log_msg log_msg(&_name, lvl);
         log_msg.raw << msg;
+        if (_formatter->do_msg_count()) {
```

Comment:
i do agree. but you had concerns about performance hit. as i have shown in my benchmark the cost of atomic increment can be neglected.
do you prefer to revert conditional increment logic?